src/GovUk.Education.ExploreEducationStatistics.Data.Services/SubjectMetaService.cs

```diff
@@ -102,30 +102,44 @@ private SubjectMetaViewModel GetSubjectMetaViewModel(Subject subject)
         private SubjectMetaViewModel GetSubjectMetaViewModelFromQuery(SubjectMetaQueryContext query)
         {
             var observations = _observationService.FindObservations(query).AsQueryable();
-
+            var locations = new Dictionary<string, ObservationalUnitsMetaViewModel>();
+            var timePeriods = new TimePeriodsMetaViewModel();
+            var filters = new Dictionary<string, FilterMetaViewModel>();
+            var indicators = new Dictionary<string, IndicatorsMetaViewModel>();
+            
             var stopwatch = Stopwatch.StartNew();
             stopwatch.Start();
 
-            var filters = query.TimePeriod != null ? GetFilters(query.SubjectId, observations, false) : new Dictionary<string, FilterMetaViewModel>();
-
-            _logger.LogTrace("Got Filters in {Time} ms", stopwatch.Elapsed.TotalMilliseconds);
-            stopwatch.Restart();
-
-            var indicators = query.TimePeriod != null ? GetIndicators(query.SubjectId) : new Dictionary<string, IndicatorsMetaViewModel>();
+            if (query.Locations == null)
+            {
+                locations = GetObservationalUnits(observations);
```

Comment:
I'm just wondering how safe this change is because it's not obvious to the caller that locations won't be returned now if locations are present in the query.

Reply:
I checked a few things on the backend and also consulted @ntsim over the way the UI works for data blocks that aren't valid to check that it's not relying on this to post the full query and expect all the meta back, which it's not, so it should be safe. Maybe some comments would help to explain what's returned in the response depending on what is included in the request?

Reply:
Yep agreed a little confusing so have added a comment as suggested - thanks Ben